src/contact.jl

```diff
@@ -227,7 +227,7 @@ end
 frame(halfspace::HalfSpace3D) = halfspace.point.frame
 
 function HalfSpace3D(point::Point3D, outward_normal::FreeVector3D)
-    T = promote_type(eltype(point), eltype(outward_normal))
+    T = promote_eltype(point, outward_normal)
```

Comment:
Module `contact.jl` uses `promote_eltype` but does not have `using Base: promote_eltype` at the top.

This can lead to the following error:
```
UndefVarError: promote_eltype not defined

Stacktrace:
 [1] RigidBodyDynamics.Contact.HalfSpace3D(::Point3D{StaticArrays.SArray{Tuple{3},Int64,1,3}}, ::FreeVector3D{StaticArrays.SArray{Tuple{3},Float64,1,3}}) at /home/henrique/.julia/packages/RigidBodyDynamics/2qxSc/src/contact.jl:230
 [2] #mechanism#1(::Bool, ::Dict{String,UnionAll}, ::RigidBodyDynamics.Contact.SoftContactModel{RigidBodyDynamics.Contact.HuntCrossleyModel{Float64},RigidBodyDynamics.Contact.ViscoelasticCoulombModel{Float64}}, ::Bool, ::Bool, ::typeof(MyPackage.ANYmalKinova.mechanism), ::Type{Float64}) at /home/henrique/git/phd-paper-code/src/ANYmalKinova.jl:60
 [3] #mechanism at ./none:0 [inlined] (repeats 2 times)
 [4] #ANYmal#37(::Bool, ::RigidBodyDynamics.Contact.SoftContactModel{RigidBodyDynamics.Contact.HuntCrossleyModel{Float64},RigidBodyDynamics.Contact.ViscoelasticCoulombModel{Float64}}, ::typeof(MyPackage.ANYmal)) at /home/henrique/git/phd-paper-code/src/MyPackage.jl:31
 [5] ANYmal() at /home/henrique/git/phd-paper-code/src/MyPackage.jl:31
 [6] top-level scope at In[7]:7
```

Reply:
@tkoolen would you like me to create a PR to fix this?

Reply:
Ah, thanks for catching this. Apparently we don't have enough test coverage of this too then. Yeah, a PR would be most welcome.